onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorQAttention.cpp

```diff
@@ -450,19 +609,72 @@ class DmlOperatorQAttention : public DmlOperator
                 maskSliceToMhaEdge.FromNodeIndex = maskSliceNodeIndex;
                 maskSliceToMhaEdge.FromNodeOutputIndex = 0;
                 maskSliceToMhaEdge.ToNodeIndex = mhaNodeIndex;
-                maskSliceToMhaEdge.ToNodeInputIndex = mhaMaskIndex;
+                maskSliceToMhaEdge.ToNodeInputIndex = MhaInputIndex::mhaMaskIndex;
                 intermediateEdges.push_back(maskSliceToMhaEdge);
             }
             else
             {
                 DML_INPUT_GRAPH_EDGE_DESC maskToMhaEdge = {};
                 maskToMhaEdge.GraphInputIndex = InputIndex::maskIndex;
                 maskToMhaEdge.ToNodeIndex = mhaNodeIndex;
-                maskToMhaEdge.ToNodeInputIndex = mhaMaskIndex;
+                maskToMhaEdge.ToNodeInputIndex = MhaInputIndex::mhaMaskIndex;
                 inputEdges.push_back(maskToMhaEdge);
             }
         }
 
+        if (unidirectional && !hasMask)
+        {
```

Comment:
Can we merge this `if ` with previous `if` for better readability?

<pre>
if(hasMask)
{
....
}
else if(unidirectional)
{
...
}
</pre>

Reply:
yes, this works
(idk why I thought keeping it separate would help with understanding these are not exclusives events, it just how we support it today.)